DuckDuckGo/AppTPHomeViewModel.swift

```diff
@@ -0,0 +1,108 @@
+//
+//  AppTPHomeViewModel.swift
+//  DuckDuckGo
+//
+//  Copyright © 2023 DuckDuckGo. All rights reserved.
+//
+//  Licensed under the Apache License, Version 2.0 (the "License");
+//  you may not use this file except in compliance with the License.
+//  You may obtain a copy of the License at
+//
+//  http://www.apache.org/licenses/LICENSE-2.0
+//
+//  Unless required by applicable law or agreed to in writing, software
+//  distributed under the License is distributed on an "AS IS" BASIS,
+//  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+//  See the License for the specific language governing permissions and
+//  limitations under the License.
+//
+
+import Foundation
+import CoreData
+import Persistence
+import Core
+import NetworkExtension
+
+class AppTPHomeViewModel: ObservableObject {
+    @Published public var blockCount: Int32 = 0
+    @Published public var appTPEnabled: Bool = false
+    
+    private let appTPDatabase: CoreDataDatabase
+    private let context: NSManagedObjectContext
+    private var firewallManager: FirewallManaging
+    
+    private var timer: Timer?
```

Comment:
I thought about the timer approach some more – even though we aren't using a persistent results controller, we should be good to use a [remote change notification](https://github.com/duckduckgo/iOS/blob/bf7cac95dd97c8a91a6f2f8be6b30de562d3f62b/Core/AppTrackingProtectionListViewModel.swift#L181) to listen for changes to the database and refresh the fetch request in response.

Reply:
I'll try this. I wasn't sure if this would do what I needed since in the other example it only dealt with merging transactions.